### ISSUE TITLE
Can we remove some of the steps from the upgrade instructions?

### ISSUE THREAD

**Issue by seancolsen** [civicrm/civicrm-sysadmin-guide#52]
There are a lot of steps to upgrade CiviCRM! Can we reduce them? Maybe some of these steps are just lingering in the docs from years past and not really needed any longer?

I'm using the Drupal upgrade instructions as an example here, but some of the sections are in other upgrade pages (especially Backdrop).

Here are some steps that I wonder if we still need: 

* [Ensure that CiviCRM is not your Drupal homepage](https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal7/#ensure-that-civicrm-is-not-your-drupal-homepage)
    * And [Restore CiviCRM as the Drupal homepage if appropriate](https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal7/#restore-civicrm-as-the-drupal-homepage-if-appropriate)
* [Disable all CiviCRM integration modules](https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal7/#disable-all-civicrm-integration-modules)
    * And [Re-enable compatible CiviCRM integration modules](https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal7/#re-enable-compatible-civicrm-integration-modules)
    * @MegaphoneJon [mentioned](https://github.com/civicrm/civicrm-sysadmin-guide/pull/51#issuecomment-342181405) that he thought this one might be irrelevant now
* [Clear cached files](https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal7/#clear-cached-files)
* [Clear the views cache](https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal7/#clear-the-views-cache)

Can others weigh in with thoughts here?

**Comment by michaelmcandrew** [comment]
My recollection, last time I looked at the steps, were that they were VERY over-cautious, by which I mean, 95-99% of upgrades would work fine with a very paired down set of instructions but that these might affect a handful of installations (why or how, I am not sure).

Might it make sense to include some of them in a section that deals with 'if your upgrade does not go as planned, these might help'.

I feel like we need to be able to explain *why* the steps are important if we are going to include them in the instructions. If we don't know why, then we are just cargo-culting

A list of 'things that might help if your upgrade doesn't go as planned but we don't know why exactly' seems like a very unsatisfactory title for a heading, but it could be sugar-coated with a 'Needs review' or something similar.

**Comment by totten** [comment]
I like @michaelmcandrew's suggestion. Emphasize the *typical* upgrade process first, and provide  the oddball contingencies/tips/tricks second.

**Comment by freephile** [comment]
FWIW, here is my [upgrade playbook](https://wiki.freephile.org/wiki/CiviCRM/WordPress#Upgrading) that I used successfully last night to go from 4.7.16 to 5.3.0 on a WordPress instance. I know this isn't the kind of upgrade documentation that a typical user would find helpful... but for experienced sysadmins, I love having a quick 'recipe'.
```bash
sudo su -
/home/greg/bin/backup.db.sh wordpress
# set some variables
docroot=/var/www/html
pluginroot="$docroot/wp-content/plugins"
config=/var/www/html/wp-content/uploads/civicrm/civicrm.settings.php
templates=/var/www/html/wp-content/plugins/files/civicrm/templates_c/
# make sure you have a backup of your civicrm.settings.php file
cp $config ~/civicrm.settings.php.bak
backup=/tmp/html.backup
mkdir $backup && chown www-data:www-data $backup
sudo -u www-data cp -ar /var/www/html $backup
package=https://download.civicrm.org/civicrm-5.3.0-wordpress.zip
wget $package
rm -rf $pluginroot/civicrm && sudo -u www-data unzip $package -d $pluginroot
rm -rf $templates/
# upgrade the database by visiting the URL in your browser
/admin.php?page=CiviCRM&q=civicrm/upgrade&reset=1

# test and then cleanup or revert
rm $(basename $package)
rm ~/civicrm.settings.php.bak
rm -rf $backup
```

**Comment by eileenmcnaughton** [comment]
@freephile you can also use the wp command line tool rather than using the url to do the upgrade - something like

wp civicrm-upgrade-db

**Comment by freephile** [comment]
Thanks @eileenmcnaughton 

I actually never used that approach until yesterday... worked like a charm on my **Drupal** site (using the drush cli).  Originally I fashioned my Civi on WordPress process after my "traditional" [Civi on Drupal process](https://wiki.freephile.org/wiki/CiviCRM#Upgrading).  Yesterday, when I had to upgrade a Civi on Drupal system, I used ` drush civicrm-upgrade --tarfile=./sites/all/modules/contrib/civicrm-5.1.2-drupal.tar.gz --backup-dir=/tmp/`  But, I didn't know the `wp` cli has the same civicrm capabilities. Is there a plugin that you have to add to `wp`? I didn't yet review https://civicrm.org/blog/totten/developers-revising-the-civicrm-installer-library-cli-wordpress but that looks like it could be what you're talking about? (And I don't mean to go off-topic, but this is about "The Upgrade Process".

**Comment by eileenmcnaughton** [comment]
try entering 

```
wp-cli civicrm --help
```

**Comment by MikeyMJCO** [comment]
Resolved with #134
